easybuild/easyblocks/o/octave.py

```diff
@@ -0,0 +1,88 @@
+##
+# Copyright 2009-2017 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Octave, implemented as an easyblock
+
+@author: Lekshmi Deepu (Juelich Supercomputing Centre)
+@author: Kenneth Hoste (Ghent University)
+"""
+import os
+
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.run import run_cmd
+
+
+EXTS_FILTER_OCTAVE_PACKAGES = ("octave --eval 'pkg list' | grep %(ext_name)s", '')
+
+
+class EB_Octave(ConfigureMake):
+    """Support for building/installing Octave."""
+
+    def configure_step(self):
+        """Custom configuration procedure for Octave."""
+
+        self.cfg.update('configopts', '--with-blas="$LIBBLAS" --with-lapack="$LIBLAPACK"')
```

Comment:
Mmmmh, I think we should check first if `$LIBBLAS` and `$LIBLAPACK` are defined, to don't pass inadvertently `--with-blas= --with-lapack=` to `configopts`, in case someone messes up and tries to install it in a toolchain without BLAS and LAPACK

Reply:
OK, will do.